packages/@aws-cdk/aws-lambda/lib/code.ts

```diff
@@ -175,9 +179,65 @@ export class AssetCode extends Code {
       this.asset!.addResourceMetadata(resource, 'Code');
     }
 
-    return  {
+    return {
       s3Bucket: this.asset!.s3BucketName,
       s3Key: this.asset!.s3ObjectKey
     };
   }
 }
+
+export interface CodePipelineCodeProps {
+  bucketNameParam?: cdk.CfnParameter;
+
+  bucketKeyParam?: cdk.CfnParameter;
+}
+
+export class CodePipelineCode extends Code {
+  public readonly isInline = false;
+  private _bucketNameParam?: cdk.CfnParameter;
+  private _bucketKeyParam?: cdk.CfnParameter;
+
+  constructor(props: CodePipelineCodeProps = {}) {
+    super();
+
+    this._bucketNameParam = props.bucketNameParam;
+    this._bucketKeyParam = props.bucketKeyParam;
+  }
+
+  public bind(construct: cdk.Construct) {
+    if (!this._bucketNameParam) {
+      this._bucketNameParam = new cdk.CfnParameter(construct, 'LambdaSourceBucketNameParameter', {
+        type: 'String',
+      });
+    }
+
+    if (!this._bucketKeyParam) {
+      this._bucketKeyParam = new cdk.CfnParameter(construct, 'LambdaSourceBucketKeyParameter', {
+        type: 'String',
+      });
+    }
+  }
+
+  public _toJSON(_?: cdk.CfnResource): CfnFunction.CodeProperty {
+    return {
+      s3Bucket: this.bucketNameParam.stringValue,
+      s3Key: this.bucketKeyParam.stringValue,
+    };
+  }
+
+  public get bucketNameParam(): cdk.CfnParameter {
+    if (this._bucketNameParam) {
+      return this._bucketNameParam;
+    } else {
+      throw new Error('You have to use CodePipelineCode to create a Lambda Function before you can access the bucketNameParam property');
```

Comment:
"Pass `CodePipelineCode` to a Lambda Function before accessing..." ?

Reply:
Changed.